app/src/main/java/com/appunite/loudius/di/NetworkModule.kt

```diff
@@ -79,6 +79,31 @@ object NetworkModule {
         }
     }
 
+    @Provides
+    @Singleton
+    @BaseAPI
+    fun provideBaseHttpClient(
+        gson: Gson,
+        @BaseAPI baseUrl: String,
+        loggingInterceptor: HttpLoggingInterceptor,
+        authInterceptor: AuthInterceptor,
+        authFailureHandler: AuthFailureHandler,
+    ): HttpClient = HttpClient(OkHttp) {
+        expectSuccess = true
+        engine {
+            addInterceptor(authInterceptor)
+            addInterceptor(TestInterceptor)
+            addInterceptor(AuthFailureInterceptor(authFailureHandler))
```

Comment:
question: why can't we provide the AuthFailureInterceptor but we do provide authFailureHandler ? 🤔

Reply:
I think we can provide this :) What benefits will it give us? Also I'm not sure if `AuthFailureInterceptor` should be annotated with singleton or not.

Reply:
in this case I guess there will not be a huge benefit, because we are creating the object just in that one place, but if in the future it will be reused somewhere or will be more complicated, the provideBaseRetrofit will grow bigger. 

I also do not see a reason or benefit not to do it, they are both annotated with @Inject so it should work out of the box anyways, no additonal work needed.